interpreter/script/script.ml

```diff
@@ -17,6 +17,9 @@ and nan = CanonicalNan | ArithmeticNan
 
 type result = result' Source.phrase
 and result' =
+  | ValueResult of result_pat
+  | SimdF32x4Result of result_pat * result_pat * result_pat * result_pat
+and result_pat =
   | LitResult of Ast.literal
   | NanResult of nanop
```

Comment:
Latest commit makes this change.
I find that I have to do a nested pattern match on LitPat/NanPat in multiple places, since `Ast.literal = Values.value Source.phrase`, everytime I have a `num_pat`, I have to dig into `num_pat.it`, then match on that to get `LitPat`, then do `.it` again.
I wonder if we can have
```
and result' =
  | NumResult of num_pat'
```
Because for NumResult, the source of the pattern is the same.
It makes the type definition look a bit asymmetrical though.

Reply:
I think the match on `it` is fine. In some cases you can actually avoid it, see my comments. In some others it might be necessary to factor out handling num_pat into a separate function anyway once you implement the SIMD case.